### ISSUE TITLE
Provisioning of IAM role for Azure Stack HCI / PowerShell Deployment tool (preview)

### ISSUE THREAD

**Issue by Karl-WE** [DellGEOS/AzureStackHOLs#42]
Dear @jaromirk 

https://github.com/DellGEOS/AzureStackHOLs/tree/main/lab-guides/08-DeployAzureStackHCICluster-DeploymentTool

for this guidance, we can safely remove the creation of the RBAC (IAM) role to register the Azure Stack HCI nodes. The MS documentation is not complete / updated. I will raise a PR at a later date.

The previously required custom appear to exist in current tenants and include even more permissions than just registering.

tldr: by today, Microsoft has created and preprovisioned Azure Stack HCI IAM roles for registration and other purposes such as Azure Arc.

**Comment by jaromirk** [comment]
I need to double check. Last time the role did not have enough permissions. Thx!

**Comment by jaromirk** [comment]
We will need to wait for official Microsoft documentation. This is mine role:

```json
{
    "id": "/subscriptions/c9d71720-73a9-4d37-80b3-9f554562fa55/providers/Microsoft.Authorization/roleDefinitions/ef1fce41-cb15-417c-b08c-b0e4765aa7d3",
    "properties": {
        "roleName": "Azure Stack HCI registration role - Custom",
        "description": "Custom Azure role to allow subscription-level access to register Azure Stack HCI",
        "assignableScopes": [
            "/subscriptions/c9d71720-73a9-4d37-80b3-9f554562fa55"
        ],
        "permissions": [
            {
                "actions": [
                    "Microsoft.Resources/subscriptions/resourceGroups/read",
                    "Microsoft.Resources/subscriptions/resourceGroups/write",
                    "Microsoft.Resources/subscriptions/resourceGroups/delete",
                    "Microsoft.AzureStackHCI/register/action",
                    "Microsoft.AzureStackHCI/Unregister/Action",
                    "Microsoft.AzureStackHCI/clusters/*",
                    "Microsoft.Authorization/roleAssignments/write",
                    "Microsoft.Authorization/roleAssignments/read",
                    "Microsoft.HybridCompute/register/action",
                    "Microsoft.GuestConfiguration/register/action",
                    "Microsoft.HybridConnectivity/register/action",
                    "Microsoft.HybridCompute/machines/extensions/write",
                    "Microsoft.HybridCompute/machines/read",
                    "Microsoft.HybridCompute/machines/write",
                    "Microsoft.HybridCompute/privateLinkScopes/read",
                    "Microsoft.GuestConfiguration/guestConfigurationAssignments/read",
                    "Microsoft.HybridCompute/machines/extensions/read"
                ],
                "notActions": [],
                "dataActions": [],
                "notDataActions": []
            }
        ]
    }
}
```

**Comment by Karl-WE** [comment]
I see, I will check on my end aswell.
